.github/workflows/tests.yml

```diff
@@ -0,0 +1,78 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - master
+      - release
+  pull_request:
+    branches:
+      - '*'
+  schedule:
+    # https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onschedule
+    # > Scheduled workflows run on the latest commit on the default or base branch
+    # i.e. this can only run on master
+    - cron:  '0 11 * * *'
+
+env:
+  zip_filepath: "batch_processing_results.zip"
+
+jobs:
+  batch_processing:
+    name: Test batch_processing.sh
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout SCT
+        uses: actions/checkout@v2
+
+      # install_sct edits ~/.bashrc, but those environment changes don't get passed to subsequent steps in GH Actions.
+      # So, we filter through the .bashrc and pass the values to $GITHUB_ENV and $GITHUB_PATH.
+      # Relevant documentation: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#environment-files
+      - name: Install SCT
+        run: |
+          PIP_PROGRESS_BAR=off ./install_sct -y
+          cat ~/.bashrc | grep "export SCT_DIR" | cut -d " " -f 2 >> $GITHUB_ENV
+          cat ~/.bashrc | grep "export MPLBACKEND" | cut -d " " -f 2 >> $GITHUB_ENV
+          cat ~/.bashrc | grep "export PATH" | grep -o "/.*" | cut -d ':' -f 1 >> $GITHUB_PATH
+
+      - name: Run batch_processing.sh
+        run: |
+          ./batch_processing.sh
+
+      # We use the REST API to fetch the most recently uploaded results from batch_processing.
+      # jq and awk are used to filter the JSON response by artifact ID, which is then used to download the right .zip.
+      # Relevant documentation: https://docs.github.com/en/rest/reference/actions#artifacts
+      - name: Fetch past results
+        run: |
+          export NEWEST_ARTIFACT_ID=$(curl -H "Accept: application/vnd.github.v3+json" \
+                                           https://api.github.com/repos/neuropoly/spinalcordtoolbox/actions/artifacts |
+                                      jq '.artifacts[].id' |
+                                      awk '{for(i=1;i<=NF;i++) if($i>maxval) maxval=$i;}; END { print maxval;}')
```

Comment:
Not a problem at all, but `jq` has the capability to sort elements so you can use that directly instead of piping into`awk`:

```bash
curl /artifacts | jq '.artifacts|=sort_by(.id)|.artifacts[-1].id'
```

Reply:
Thank you for pointing this out. Truthfully, I had never used `jq` before, so I appreciate tips like this. :)